src/G2/Execution/InstTypes.hs

```diff
@@ -62,5 +65,17 @@ instType' (ng, st) i
                     ,type_env = te}
             st'' = replaceTyVar n t st'
         in
-        (ng',st'')
-    | otherwise = (ng, st)
\ No newline at end of file
+         (ng',st'')
+    | otherwise = (ng, st)
+        
+-- define a new reducer that calls your instType on your onAccept function
+instTypeRed :: (ASTContainer t Type, Monad m) => Reducer m () t
+instTypeRed  = (mkSimpleReducer
+                        (\_ -> ())
+                        (\rv s b -> return (NoProgress, [(s , rv)], b)) )
+                        {
+                         onAccept = \s b _ ->
+                         let 
+                               (b', s') = instType b s
+                         in 
+                            return (s', b')} 
```

Comment:
Why not just give `instType` the return Type `(State, Bindings)`, so that `onAccept` can be written:
```
onAccept = \s b _ -> return (instType b s)
```
?